dbt/models/qc/schema.yml

```diff
@@ -221,6 +221,244 @@ models:
                 category: sales_issues
                 description: mydec should not have >1000 sales that are not in iasworld
 
+  - name: qc.vw_ic_reference_all_non_res_land_details
+    description: '{{ doc("view_vw_ic_reference_all_non_res_land_details") }}'
+    config: &qc_report_ic_reference_tags
+      tags:
+        - qc_report_ic_reference
+    meta:
+      export_name: AllNonRes_LandDetails
+      export_template: &qc_report_ic_reference_export_template
+        start_row: 3
+        add_table: false
+      export_format:
+        columns:
+          - name: ACRES
+            index: J
+            data_type: float
```

Comment:
I'm a little confused why basically all of our fields get written to Excel as strings with `df.to_excel()`, given that all of these fields have correct types in iasWorld and we're using `unload=True` when querying them. Specifying `data_type` fixes this issue, but do you have insight into why it might be happening in the first place?

Reply:
PyAthena doesn't seem to work the way I would expect based on the docs. It seems like `unload` should perform this type conversion for us, without the need for a custom converter:

> If the unload option is enabled, the Parquet file itself has a schema, so the conversion is done to the dtypes according to that schema, and the mappings and types settings of the Converter class are not used.

However, in practice, `PandasCursor` with `unload=True` seems to always return `Object` type. However, `ArrowCursor` performs the conversion I would expect (e.g. `comdat.ovrrcnld` gets converted to `pyarrow.decimal128(10, 0)`). I suspect this may be exposing a bug in the PyAthena implementation, but I don't want to spend too much time isolating it right now.

Most likely due to the automatic conversion that the docs describe, the custom converter also doesn't seem to do anything when set on a cursor with `unload=True`. We could switch to a different cursor type, but I do think the pandas cursor with `unload=True` is the easiest one to work with in this case. I  also don't think a custom converter would substantially reduce our code even if we could get it working, because our Athena types are often ambiguous (e.g. `DECIMAL` sometimes indicates a float and sometimes indicates an integer,  and `VARCHAR` sometimes indicates a decimal stored as a string). We might save a little bit of space on the schema specification, but I don't think it's worth it for this particular application. Definitely open to other opinions though!

Reply:
Alright, let's just go with the existing setup then, as it works and is perfectly clear.